p11kit/attribute.go

```diff
@@ -78,6 +78,25 @@ func (o *Object) SetLabel(label string) {
 	})
 }
 
+// SetCertificate set certificate attributes to key object.
+// It will create the link between key and certificate.
+func (o *Object) SetCertificate(cert *x509.Certificate) error {
+	keyType, exist := o.attributeValue(attributeClass)
+	if !exist {
+		return fmt.Errorf("CKA_CLASS doesn't exist")
+	}
+	if keyType.ulong == nil || (*keyType.ulong != ckoPrivateKey && *keyType.ulong != ckoPublicKey) {
```

Comment:
nit: this could just be

```
if o.pub == nil && o.priv == nil {
    // ...
}
```

Of if you just want to record the cko attribute on the object, that works too